internal/services/automation/automation_software_update_configuration.go

```diff
@@ -0,0 +1,949 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/automation/mgmt/2020-01-13-preview/automation"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	validate3 "github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	computeValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	validate2 "github.com/hashicorp/terraform-provider-azurerm/internal/services/resource/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/suppress"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func parseTime(s string) *date.Time {
+	t, _ := time.Parse(time.RFC3339, s)
+	return &date.Time{
+		Time: t,
+	}
+}
+
+func timeString(t *date.Time) string {
+	if t == nil {
+		return ""
+	}
+	return t.Time.Format(time.RFC3339)
+}
+
+func stringSlice(input *[]string) []string {
+	if input == nil {
+		return nil
+	}
+	return *input
+}
+
+func deFloat64(ptr *float64) float64 {
+	if ptr == nil {
+		return 0
+	}
+	return *ptr
+}
```

Comment:
how come we have written these instead of using the buildin ones?

Reply:
sorry I didn't find such builtin functions under `utils` package for time and float64..

Reply:
https://github.com/hashicorp/go-azure-helpers/blob/main/lang/dates/parse.go
https://github.com/hashicorp/go-azure-helpers/blob/main/lang/pointer/from.go
https://github.com/hashicorp/go-azure-helpers/blob/main/lang/pointer/to.go

Reply:
anything that is missing should be added to utils